invenio_rdm_records/services/components/pids.py

```diff
@@ -89,6 +90,29 @@ def publish(self, identity, draft=None, record=None):
             if record_id != draft_id:
                 changed_pids[scheme] = record_pids[scheme]
 
+        if current_app.config["DATACITE_ENABLED"]:
```

Comment:
I remember vaguely a conversation about specilised pid providers, shouldn't this be implemented on the datacite provider instead?